crates/re_types/src/components/media_type_ext.rs

```diff
@@ -1,15 +1,133 @@
+
+    /// [Wavefront .obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file).
+    ///
+    /// <https://www.iana.org/assignments/media-types/model/obj>
+    pub const OBJ: &str = "model/obj";
+}
+
 impl MediaType {
     /// `text/plain`
     #[inline]
     pub fn plain_text() -> Self {
-        Self("text/plain".into())
+        Self(Self::TEXT.into())
     }
 
     /// `text/markdown`
     #[inline]
     pub fn markdown() -> Self {
-        Self("text/markdown".into())
+        Self(Self::MARKDOWN.into())
+    }
+
+    /// `model/gltf+json`
+    #[inline]
+    pub fn gltf() -> Self {
+        Self(Self::GLTF.into())
+    }
+
+    /// `model/gltf-binary`
+    #[inline]
+    pub fn glb() -> Self {
+        Self(Self::GLB.into())
+    }
+
+    /// `model/obj`
+    #[inline]
+    pub fn obj() -> Self {
+        Self(Self::OBJ.into())
+    }
+}
+
+impl MediaType {
+    #[inline]
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl MediaType {
+    /// Tries to guess the media type of the file at `path` based on its extension.
+    #[inline]
+    pub fn guess_from_path(path: impl AsRef<std::path::Path>) -> Option<Self> {
+        mime_guess::from_path(path)
+            .first_raw()
+            .map(ToOwned::to_owned)
+            .map(Into::into)
+    }
+
+    /// Tries to guess the media type of the file at `path` based on its contents (magic bytes).
+    #[inline]
+    pub fn guess_from_data(data: &[u8]) -> Option<Self> {
+        fn glb_matcher(buf: &[u8]) -> bool {
+            buf.len() >= 4 && buf[0] == b'g' && buf[1] == b'l' && buf[2] == b'T' && buf[3] == b'F'
+        }
+
+        // NOTE:
+        // - gltf is simply json, so no magic byte
+        //   (also most gltf files contain file:// links, so not much point in sending that to
+        //   Rerun for now…)
+        // - obj is simply text, so no magic byte
+
+        let mut inferer = infer::Infer::new();
+        inferer.add(Self::GLB, "", glb_matcher);
+
+        inferer
+            .get(data)
+            .map(|v| v.mime_type())
+            .map(ToOwned::to_owned)
+            .map(Into::into)
+    }
+
+    /// Returns `opt` if not `None`, otherwise tries to guess a media type using [`Self::guess_from_path`].
+    #[inline]
+    pub fn or_guess_from_path(
+        opt: Option<Self>,
+        path: impl AsRef<std::path::Path>,
+    ) -> Option<Self> {
+        opt.or_else(|| Self::guess_from_path(path))
+    }
+
+    /// Returns `opt` if not `None`, otherwise tries to guess a media type using [`Self::guess_from_data`].
+    #[inline]
+    pub fn or_guess_from_data(opt: Option<Self>, data: &[u8]) -> Option<Self> {
+        opt.or_else(|| Self::guess_from_data(data))
+    }
+}
+
+impl std::str::FromStr for MediaType {
+    type Err = <mime_guess::Mime as std::str::FromStr>::Err;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mime: mime::Mime = s.parse()?;
+        Ok(Self(mime.to_string().into()))
```

Comment:
I think this is a bad idea.

`MediaType` is declared as `struct MediaType(pub crate::datatypes::Utf8)` which means we make no guarantees that what it contains is sane, yet here in one single place we validate it. This is inconsistent.

I think it would make more sense to add a member function for sanity checking, i.e.:

```rs
impl MediaType {
    /// Returns an error if the contained string does not strictly conform to RFC2046 (formerly known as MIME types).
    pub fn validate(&self) -> Result<InvalidMediaType, ()> {
}
```

Though I'm not sure when or why we would call it.